File: php-phpunit/test/Session5_WordCounterRatioTest.php
<?php

use \Wordcount\WordCounter;

/**
 * Session 5: WordCounterTableTest - parameterised/table driven tests.
 * See https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers
 */
class Session5_WordCounterRatioTest extends \PHPUnit_Framework_TestCase {

    // TODO Add the needed annotations to run this test with all examples.

    // We want to test more corner cases for the ratio. Here is a table of test cases.
    static function tableData() {
        return [
            // instructor shows this table data in slides
            [ "green", "green", 1.0 ], //
            [ "green bar green", "green", 0.66 ], //
            [ "green bar green bar", "green", 0.5 ], //
            [ "green bar green", "bar", 0.33 ] //
        ];
    }

    /**
     * @test
     * @dataProvider tableData
     */
    function shouldReturnRatioOfGivenWord($sentence, $word, $expectedRratio) {
        $counter = new WordCounter($sentence);
        $this->assertEquals($expectedRratio, $counter->ratioOf($word), "", 0.01);
    }
}

File: csharp-xunit/WordCountTests/Session5_WordCounterRatioTest.cs
using Org.Codecop.WordCount;
using System.Collections.Generic;
using Xunit;

namespace Org.Codecop.WordCount.Tests
{
    /// <summary>Session 5: WordCounterRatioTest - parameterised/table driven tests.</summary>
    /// <seealso>"http://hamidmosalla.com/2017/02/25/xunit-theory-working-with-inlinedata-memberdata-classdata/"</seealso>
    public class Session5_WordCounterRatioTest
    {

        // TODO Add the needed annotations to run this test with all examples.
        // We want to test more corner cases for the ratio. Here is a table of test cases.
        [Theory]
        [InlineData("green", "green", 1.0)]
        [InlineData("green bar green", "green", 0.66)]
        [InlineData("green bar green bar", "green", 0.5)]
        [InlineData("green bar green", "bar", 0.33)]
        public void ShouldReturnRatioOfGivenWord(string sentence, string word, double expectedRatio)
        {
            var counter = new WordCounter(sentence);
            Assert.Equal(expectedRatio, counter.RatioOf(word), 3); // keep
        }
    }
}
